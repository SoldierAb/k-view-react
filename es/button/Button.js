import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["children", "type"];
import React from 'react';
import t from 'prop-types';
var prefixCls = 'k';
var types = {
  primary: '#5352ED',
  success: '#2ED573',
  warning: '#FFA502',
  error: '#FF4757'
};

var Button = function Button(_ref) {
  var children = _ref.children,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'primary' : _ref$type,
      rest = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement("button", _extends({
    className: "".concat(prefixCls, "-button-container"),
    style: {
      background: types[type]
    }
  }, rest), children);
};

Button.propTypes = {
  type: t.oneOf(['primary', 'success', 'warning', 'error'])
};
export default Button;