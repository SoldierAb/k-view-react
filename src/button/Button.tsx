import React from 'react'
import type { ReactFragment } from 'react'
import t from 'prop-types'

export interface ButtonProps {
    /**
     * @description Button 类型
     * @default 'primary'' 
     */
    type?: 'primary' | 'success' | 'warning' |'error';
    children?: ReactFragment
}

const prefixCls = 'k';

export type ButtonTypes = Record<Required<ButtonProps>['type'], string>

const types: ButtonTypes = {
  primary: '#5352ED',
  success: '#2ED573',
  warning: '#FFA502',
  error: '#FF4757',
}

const Button: React.FC<ButtonProps> = ({ children, type = 'primary', ...rest}) => {
    return <button className={`${prefixCls}-button-container`} style={{background: types[type]}} {...rest}>{children}</button>
}

Button.propTypes = {
    type: t.oneOf(['primary', 'success', 'warning', 'error'])
}

export default Button