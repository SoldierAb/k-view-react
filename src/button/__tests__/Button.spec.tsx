import React from 'react'
import { render } from '@testing-library/react'
import Button from '../Button'

describe('Button',  ()=>{
    test('render', ()=>{
        const { container } = render(<Button>primary</Button>)
        expect(container).toMatchSnapshot()
    })
    test('type render correctly', ()=>{
        const types: any[] = ['primary','success','warning','error']
        const { getByText }  = render(
            <>
                {
                    types.map(ty => <Button type={ty} key={ty}>{ty}</Button>)
                }
            </>
        )
        types.forEach(ty => expect(getByText(ty)).toMatchSnapshot())
    })
})