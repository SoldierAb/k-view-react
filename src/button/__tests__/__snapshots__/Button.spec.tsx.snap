// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<div>
  <button
    class="k-button-container"
    style="background: rgb(83, 82, 237);"
  >
    primary
  </button>
</div>
`;

exports[`Button type render correctly 1`] = `
<button
  class="k-button-container"
  style="background: rgb(83, 82, 237);"
>
  primary
</button>
`;

exports[`Button type render correctly 2`] = `
<button
  class="k-button-container"
  style="background: rgb(46, 213, 115);"
>
  success
</button>
`;

exports[`Button type render correctly 3`] = `
<button
  class="k-button-container"
  style="background: rgb(255, 165, 2);"
>
  warning
</button>
`;

exports[`Button type render correctly 4`] = `
<button
  class="k-button-container"
  style="background: rgb(255, 71, 87);"
>
  error
</button>
`;
